doc: Add troubleshooting section (#620)
arista-netdevops-community/anta: docs/faq.md

# Frequently Asked Questions (FAQ)

## `<Foo>Timeout` error in the logs
???+ faq "`Timeout` error in the logs"

>     When running the `anta`, you receive `<Foo>Timeout` errors in the logs (could be ReadTimeout, WriteTimeout, ConnectTimeout, PoolTimeout, ... you name it Timeout).

Reviewer: ```suggestion
    When running ANTA, you can receive `<Foo>Timeout` errors in the logs (could be ReadTimeout, WriteTimeout, ConnectTimeout, PoolTimeout). More details on the timeouts of the underlying library are available here: https://www.python-httpx.org/advanced/timeouts/
```

    This might be due to the time the host on which ANTA is run takes to reach the target devices (for instance if going through firewalls, NATs, ...) or when a lot of tests are being run at the same time on a device (eAPI has a queue mechanism to avoid exhausting EOS resources because of a high number of simultaneous eAPI requests).

    ### Solution

    Use the `timeout` option. As an example for the `nrfu` command:

    ```bash
    anta nrfu --enable --username username --password arista --inventory inventory.yml -c nrfu.yml --timeout 50 text
    ```

    The previous command set a couple of options for ANTA NRFU, one them being the `timeout` command, by default, when running ANTA from CLI, it is set to 30s.

>     The timeout is increased to 50s to allow device and ANTA to wait on each other a little longer.

Reviewer: ```suggestion
    The timeout is increased to 50s to allow ANTA to wait for API calls a little longer.
```
